aten/src/ATen/cuda/CUDAContext.h

```diff
@@ -41,6 +41,14 @@ inline int64_t getNumGPUs() {
     return c10::cuda::device_count();
 }
 
+/**
+ * In some situations, you may have compiled with CUDA, but no CUDA
+ * device is actually available.  Test for this case using is_available().
+ */
+inline bool is_available() {
```

Comment:
Is it ok that this logic is different from the [python implementation of is_available()](https://github.com/pytorch/pytorch/blob/b15242f70cce10c0a0f8c36b690ac2481c51814f/torch/cuda/__init__.py#L45-L50) ?

Reply:
Hmm, I guess it would be good to test the driver version here too. I wonder where that code lives...

Reply:
It's [here](https://github.com/pytorch/pytorch/blob/c5cc1e3ab28380d1e1a4a1a0db8a295bd45c0bd8/torch/csrc/cuda/Module.cpp#L91-L99) but the check is not what I expected. Maybe this test is not really useful in this case, or it should be done during the initialization phase?

Reply:
No, it's good to have the check here. I think I know how to do it, fix incoming.